mirrord/agent/iptables/src/lib.rs

```diff
@@ -238,6 +217,28 @@ where
         Ok(Self { redirect })
     }
 
+    /// detect if the IP tables on the target are dirty: this is required since we use static
+    /// IP tables chain names, and they are in danger of conflicting with each-other
+    ///
+    /// returns `true` if iptables are clean and the agent can proceed, other returns `false`
+    #[tracing::instrument(level = Level::TRACE, skip(ipt) ret, err)]
+    pub async fn ensure_iptables_clean(ipt: IPT) -> IPTablesResult<bool> {
+        let ipt = Arc::new(ipt);
+        tracing::trace!(
+            "list rules in {IPTABLES_TABLE_NAME}: {:?}",
+            ipt.list_table(IPTABLES_TABLE_NAME)?
+        );
+        let rules = ipt.list_table(IPTABLES_TABLE_NAME)?;
+        if rules.iter().any(|rule| {
+            [IPTABLE_PREROUTING, IPTABLE_MESH, IPTABLE_STANDARD]
+                .iter()
+                .any(|&chain| rule.contains(&chain.to_string()))
```

Comment:
```suggestion
                .any(|chain| rule.contains(*chain))
```
Sth like this should work